### PR TITLE
bpo-40994: Ungroup items in collections.abc documentation for improved clarity

### PR DESCRIPTION
**Pull request by sydneypemberton1986**
Use a less surprising document structure.

https://bugs.python.org/issue40994

Automerge-Triggered-By: @csabella

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@sydneypemberton1986

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Doc/library/collections.abc.rst [hunk 1]
```diff
@@ -98,12 +98,20 @@ ABC                        Inherits from          Abstract Methods        Mixin
...
-           Callable
 
-   ABCs for classes that provide respectively the methods :meth:`__contains__`,
-   :meth:`__hash__`, :meth:`__len__`, and :meth:`__call__`.
+   ABCs for classes that provide method :meth:`__hash__`.
+
+.. class:: Hashable
+
+   ABCs for classes that provide method :meth:`__len__`.
+
```
Comment (iritkatriel) [inline Doc/library/collections.abc.rst:107]: ABC, not ABCs

**Comment by iritkatriel** [comment]
I think the title should be, instead of "Clarify documentation using 'respectively' construct", something that says more clearly where the change was made. Something like "Ungroup items in collections.abc documentation for improved clarity".

File: Doc/library/collections.abc.rst [hunk 2]
```diff
@@ -98,12 +98,20 @@ ABC                        Inherits from          Abstract Methods        Mixin
...
-           Hashable
-           Sized
-           Callable
 
-   ABCs for classes that provide respectively the methods :meth:`__contains__`,
-   :meth:`__hash__`, :meth:`__len__`, and :meth:`__call__`.
+   ABCs for classes that provide method :meth:`__hash__`.
+
+.. class:: Hashable
+
```
Comment (iritkatriel) [inline Doc/library/collections.abc.rst:105]: It seems off - you have __len__ under hashable, __call__ under Sized?

**Comment by sydneypemberton1986** [comment]
@iritkatriel Thank you so much for the prompt review. I was tangling with the first time setup of a Python PR and I made some pretty bonehead errors there. I've addressed your feedback and am working on straightening out the CLA. I know I signed it but it's not getting associated with my account somehow.

**Comment by iritkatriel** [comment]
Thank you for addressing the comments. I think the CLA can take a day or two. In the meantime, it would be tidier to squash the two commits into one, so that the commit log is clean.

**Comment by sydneypemberton1986** [comment]
Excellent. I will rebase them.

File: Doc/library/collections.abc.rst [hunk 3]
```diff
@@ -98,12 +98,20 @@ ABC                        Inherits from          Abstract Methods        Mixin
...
+   ABC for classes that provide method :meth:`__hash__`.
+
+.. class:: Sized
+
+   ABC for classes that provide method :meth:`__len__`.
+
+.. class:: Callable
+
+   ABC for classes that provide method :meth:`__call__`.
```
Comment (iritkatriel) [inline Doc/library/collections.abc.rst:115]: Actually I just noticed - the wording is inconsistent with the other items, like, just below it says:

"ABC for classes that provide the :meth:`__iter__` method."

and not

"ABC for classes that provide method :meth:`__iter__`."

**Comment by sydneypemberton1986** [comment]
I have my Github username configured as `@sydneypemberton1986` on my B.P.O account.
I'm not exactly sure what counts as my B.P.O. username when signing the CLA. I used the federated login with Google. 
I've tried a few variations, maybe I just need to wait.

File: Doc/library/collections.abc.rst (hunk 3 above)
Comment (sydneypemberton1986) [reply Doc/library/collections.abc.rst:115]: Excellent eye. I've updated it to be consistent.

**Review by iritkatriel — APPROVED** [review]

**Review by eamanu — APPROVED** [review]
Hi, I like this improve, LGTM

**Review by rhettinger — APPROVED** [review]

**Comment by csabella** [comment]
@sydneypemberton1986, looking at the user screen on bugs.python.org, I believe you need to remove the '@' at the beginning of the Github name for the bot to be able to match to it.  Thanks!

**Comment by csabella** [comment]
I'm going to close and reopen this to retrigger Travis CI.

**Comment by csabella** [comment]
@sydneypemberton1986 thank you for the contribution and welcome to CPython!  🎉 🎉 🎉 

@iritkatriel, @eamanu, and @rhettinger, thank you for the reviews!

**Pull request merged**